save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt

```diff
@@ -135,11 +138,13 @@ val requestModalHandler: FC<PropsWithChildren> = FC { props ->
         arrayOf(statusContext)
     ) { statusContext }
 
-    val reactNode = if (modalState.status == 404.toShort()) {
+    val reactNode = if (modalState.status == 404.toShort() && modalState.isNeedRedirect) {
```

Comment:
It seems a little strange, that the flag is set in the response handler if `!response.ok`, however, it's only used if status is 404 as well. It would be more clear if the logic would be in a single place. For example, set flag in response handler only if status is 404, and then, in the render method, if the flag is set, render FallbackView with text based on current status in the context.

Reply:
Fixed

Reply:
Error status on the page is still always `404`, should we use status from context?